### DISCUSSION TITLE
Accounting for biases in manipulated variables using set_variable timevarying_parameter

### DISCUSSION THREAD

**Discussion by pavaninguva** [do-mpc/do-mpc#341] [Q&A]
Hi, I was wondering if it is possible to account for biases in a manipulated variable using the set_variable(var_type="timevarying_parameter")? 

Using the batch bioreactor as an example (https://www.do-mpc.com/en/v4.1.1/example_gallery/batch_reactor.html), the manipulated variable is the feedflow rate u. Supposing that there was a bias (either transient or steady-state) where u_actual is say 10% less than u_calculated (i.e., the value of u determined from the optimizer), is there a way to implement this to see the controller performance. My understanding is that we need to specify the values of the timevarying_parameter explicitly which would preclude this as we would need to specify u_actual as a function of u_calculated. 

Thanks for the help!

**Comment by 4flixt** [comment]
Hi @pavaninguva,

there are several straight forward solutions to this:
1. Create an additional simulation model where your applied input is 0.9*u
2. Use a parameter in the model which can be multiplied with the input
3. Use a time-varying parameter in the model which can be multiplied with the input.

You can find [here](https://www.do-mpc.com/en/latest/FAQ.html#time-varying-parameters) some information on time-varying parameter vs. parameters.

Best,
4flixt
